azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -954,6 +952,7 @@ def _run_op(self, op):
                 )
                 self.waiting_connect_ops.append(op)
             else:
+                # TODO: we are missing test cases for this condition
```

Comment:
not interefering with user-initiated connects is a pretty big breaking change because autoconnect affects most of our APIs and this is considered a user-initiated change.  This means that any API could fail in cases where it previously did not.  The behavior difference would be hard to predict because a user won't know if any particular call to send_message will start a connection (and maybe fail) or use an existing connection (and probably succeed)